tests/test_api.py

```diff
@@ -1710,3 +1711,78 @@ def test_reactor_config_secret(self):
             }
         }
         assert expected_secret in secrets
+
+    @pytest.mark.parametrize(('platform', 'release', 'platforms',  # noqa
+                              'worker', 'orchestrator',
+                              'arrangement_version', 'raises_exception'), [
+        # worker build
+        ("plat", 'rel', None, True, False, 1, False),
+        ("plat", 'rel', None, True, False, None, True),
+        # orchestrator build
+        (None, None, 'platforms', False, True, 1, True),
+        (None, None, 'platforms', False, True, None, False),
+        # prod build
+        (None, None, None, False, False, 1, True),
+        (None, None, None, False, False, None, False),
+    ])
+    def test_arrangement_version(self, osbs, platform, release, platforms,
+                                 worker, orchestrator,
+                                 arrangement_version, raises_exception):
+        class MockArgs(object):
+            def __init__(self, platform, release, platforms, arrangement_version,
+                         worker, orchestrator):
+                self.platform = platform
+                self.release = release
+                self.platforms = platforms
+                self.arrangement_version = arrangement_version
+                self.worker = worker
+                self.orchestrator = orchestrator
+                self.scratch = None
+
+        expected_kwargs = {
+            'platform': platform,
+            'scratch': None,
+            'platforms': platforms,
+            'release': release,
+            'git_uri': None,
+            'git_ref': None,
+            'git_branch': None,
+            'user': None,
+            'tag': None,
+            'target': None,
+            'architecture': None,
+            'yum_repourls': None,
+        }
+        if worker:
+            expected_kwargs['arrangement_version'] = arrangement_version
+
+        print(expected_kwargs)
+        if worker and not raises_exception:
+            (flexmock(osbs)
+                .should_receive("create_worker_build")
+                .times(1)
+                .with_args(**expected_kwargs)
+                .and_raise(StopIteration))
+
+        if orchestrator and not raises_exception:
+            (flexmock(osbs)
+                .should_receive("create_orchestrator_build")
+                .times(1)
+                .with_args(**expected_kwargs)
+                .and_raise(StopIteration))
+
+        if not worker and not orchestrator and not raises_exception:
+            (flexmock(osbs)
+                .should_receive("create_prod_build")
+                .times(1)
+                .with_args(**expected_kwargs)
+                .and_raise(StopIteration))
```

Comment:
Oh, okay, that makes sense, although we should use a custom exception instead of `StopIteration` - so that if any function using that wouldn't break the test